src/libraries/System.Reflection.DispatchProxy/tests/DispatchProxyTests.cs

```diff
@@ -557,6 +557,28 @@ public static void Proxy_Declares_Interface_Properties(bool useGenericCreate)
             Assert.NotNull(propertyInfo);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void Proxy_Declares_Interface_Static_Virtual_Properties(bool useGenericCreate)
+        {
+            TestType_IStaticVirtualPropertyService proxy = CreateHelper<TestType_IStaticVirtualPropertyService, TestDispatchProxy>(useGenericCreate);
+            PropertyInfo propertyInfo = proxy.GetType().GetTypeInfo().GetDeclaredProperty(nameof(TestType_IStaticVirtualPropertyService.TestProperty));
+            Assert.NotNull(propertyInfo);
+            Assert.True((propertyInfo.GetMethod!.Attributes & MethodAttributes.Static) != 0);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void Proxy_Declares_Interface_Static_Virtual_Methods(bool useGenericCreate)
+        {
+            TestType_IStaticVirtualMethodService proxy = CreateHelper<TestType_IStaticVirtualMethodService, TestDispatchProxy>(useGenericCreate);
+            MethodInfo methodInfo = proxy.GetType().GetTypeInfo().GetDeclaredMethod(nameof(TestType_IStaticVirtualMethodService.TestMethod));
+            Assert.NotNull(methodInfo);
+            Assert.True((methodInfo.Attributes & MethodAttributes.Static) != 0);
+        }
+
```

Comment:
We also need test that invoke the method and that the verify that the proxy received correct values of the method arguments.

Reply:
> We also need tests that invoke the method and that verify that the proxy received correct values of the method arguments.

Do you mean to test the result of e.g. this?
```C#
private static string TestStaticMethodInvocation<T>(T proxy)
    where T : TestType_IStaticVirtualMethodService
    => T.TestMethod();
```
After all, the method cannot be invoked on the proxy instance itself since it's static.

Reply:
Good point!

So what would be the behavior of calling `methodInfo.Invoke(...)` in this test case?

Reply:
> Good point!
> 
> So what would be the behavior of calling `methodInfo.Invoke(...)` in this test case?

Unless I'm doing something wrong, this results in:
```
System.InvalidProgramException : Common Language Runtime detected an invalid program.
        Stack Trace:
             at generatedProxy_13.TestMethod(String)
             at InvokeStub_generatedProxy_13.TestMethod(Object, Span`1)
             at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

Being able to call a static interface method on an instance of the interface should not be possible, so something like this seems expected.

The point is not to be able to call this method on the proxy instance but rather to be able to generate the proxy so one can call the non-static members of the interface, which is not possible atm since it fails to generate the proxy at all.